sycl/include/sycl/detail/util.hpp

```diff
@@ -42,6 +42,8 @@ template <typename T> struct TempAssignGuard {
   TempAssignGuard(T &fld, T tempVal) : field(fld), restoreValue(fld) {
     field = tempVal;
   }
+  TempAssignGuard(const TempAssignGuard<T> &) = delete;
+  TempAssignGuard operator=(const TempAssignGuard<T> &) = delete;
```

Comment:
Whole point of this class is to use the constructor to temporary assign a value. Since the copy assignment/copy constructor cant take an extra assignment for the temp value to assign these functions are not useful.